languages/emf-metamodel/src/main/java/de/jplag/emf/normalization/TokenOccurenceVector.java

```diff
@@ -0,0 +1,39 @@
+package de.jplag.emf.normalization;
+
+import java.util.List;
+
+/**
+ * A vector for the occurrence frequency of different token types. The vector is padded with zeroes beyond its original
+ * size. The vector content cannot be changed after its creation.
+ */
+public class TokenOccurenceVector {
+    private final List<Double> originalVector;
+
+    /**
+     * Creates a zero-padded token occurrence vector.
+     * @param originalVector specifies the occurrence frequency values for the vector.
+     */
+    public TokenOccurenceVector(List<Double> originalVector) {
+        this.originalVector = originalVector;
+    }
+
+    /**
+     * Return a occurrence frequency value of the vector at the specified.
+     * @param index is the specified index.
+     * @return the occurrence frequency value or zero if the index is beyond the size of the vector.
+     */
+    public double get(int index) {
```

Comment:
Would it not be easier to extends the List class and overwrite the get method.
TokenOccurenceVector just stores an immutable list with custom get logic

Reply:
Would be possible, however, I would say delegation > inheritance here. Moreover, it could be interpreted as violating the contract of `List`.